app/src/main/java/com/example/mvpdemologinsreen/model/source/UserRepository.java

```diff
@@ -0,0 +1,70 @@
+package com.example.mvpdemologinsreen.model.source;
+
+import android.content.Context;
+
+import com.example.mvpdemologinsreen.model.User;
+import com.example.mvpdemologinsreen.model.source.local.UserRunnable;
+import com.example.mvpdemologinsreen.model.source.local.UserDatabase;
+
+import java.util.ArrayList;
+
+public class UserRepository implements UserDataSource, onDataLoadingCallback<ArrayList<User>> {
+
+
+    private UserDatabase mUserDatabase;
+    private onSetMessageCallback mOnSetMessageCallback;
+    private ArrayList<User> mListUser;
+    private Context mContext;
+    private Thread mUserThread;
+
+    public UserRepository(Context context) {
+        mUserDatabase = UserDatabase.getInstance(context);
+        mContext = context;
+        getListUserFromDatabase();
+    }
+
+
+    @Override
+    public boolean addUser(User user) {
+        if (mUserDatabase != null && mUserDatabase.addUser(user)) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void getListUserFromDatabase() {
+        if (mUserThread == null) {
+            mUserThread = new Thread(new UserRunnable(mContext, this));
+            mUserThread.start();
+        } else mUserThread.run();
+    }
+
+    @Override
+    public void registerAccount(String email, String password) {
+
+        String message ;
+        if(!mUserDatabase.hasEmail(email) && addUser(new User(email,password))) {
+            message =  "Register Successfully";
+        } else message =  "Password must more than 6 character or Email has been existed";
```

Comment:
```suggestion
        private static final String MSG_ERROR =  "Password must more than 6 character or Email has been existed";
         private static final String MSG_SUCCESSFUL =  "Register Successfully";
```

Reply:
Chắc Tài nhầm gì đó chứ cái này ở trong method ko dùng private static dc. Việc đặt tên thành 2 message khác nhau thế này thì rõ ràng là phải gọi lại 2 phương thức setSendMessageToPresenter trong một phương thức. Tôi thấy vậy là không hợp lý lắm.

Reply:
Ý Tài là em nên khai báo 1 hằng số private static final String MSG_SUCCESSFUL =  "Register Successfully";
và ở đây gọi ntn
```suggestion
        } else message =  MSG_SUCCESSFUL;
```

Reply:
như thế này thì làm sao a biết ntn lkaf login thành công và như thế nào là thất bại em nhỉ?

Reply:
Em hiểu rồi ạ. 
rep comment : "như thế này thì làm sao a biết ntn lkaf login thành công và như thế nào là thất bại em nhỉ?"
Lúc đầu demo em chỉ có ý định nhận message từ hệ thống và toast lên cho người dùng xem thôi ạ